errors/next-prerender-current-time.md

```diff
@@ -0,0 +1,133 @@
+---
+title: Cannot access `Date.now()`, `Date()`, or `new Date()` while prerendering
+---
+
+#### Why This Error Occurred
+
+A function is calling a `Date` API that provides the current time. The current time is not prerenderable and must be excluded. The correct solution depends on your use case. You can find more information below about possible ways to resolves this issue.
+
+#### Possible Ways to Fix It
+
+If you are using the current time for performance tracking consider using `performance.now()` and related APIs instead.
+
+Before:
+
+```jsx filename="app/page.js"
+export default async function Page() {
+  const start = Date.now();
+  const data = await somethingSlow();
+  const end = Date.now();
+  console.log(`somethingSlow took ${end - start} milliseconds to complete`)
+  return ...
+}
+```
+
+After:
+
+```jsx filename="app/page.js"
+export default async function Page() {
+  const start = performance.now();
+  const data = await somethingSlow();
+  const end = performance.now();
+  console.log(`somethingSlow took ${end - start} milliseconds to complete`)
+  return ...
+}
+```
+
+If you want the current time to be reactive so it can change and update consider moving this usage to a Client Component. Note that Server Side Rendering timestamps is also problematic because hydration will fail so you must only set the time after hydration has completed using `useEffect()`.
```

Comment:
is that a new recommendation? hydration wouldn’t fail if you use the suppress prop which used to be the recommendations for timestamps. and it’s not uncommon to want timestamps to be reflected in SSR render (even if they get replaced on hydration)